libsrc/hyperion/LinearColorSmoothing.cpp

```diff
@@ -175,11 +175,40 @@ void LinearColorSmoothing::queueColors(const std::vector<ColorRgb> & ledColors)
 void LinearColorSmoothing::componentStateChange(const hyperion::Components component, const bool state)
 {
 	if(component == hyperion::COMP_SMOOTHING)
+	{
 		setEnable(state);
+	}
+
+	if(component == hyperion::COMP_LEDDEVICE)
+	{
+		setEnableDevice(state);
+	}
+}
+
+void LinearColorSmoothing::setEnableDevice(bool enable)
+{
+	// If LedDevice is disabled
+	if ( !enable )
+	{
+		bool switchable = _hyperion->getActiveDevice()->isSwitchable();
+
+		// If LEDDevice is switchable, clear output queue and stop writing to device
+		if ( switchable )
+		{
+			QMetaObject::invokeMethod(_timer, "stop", Qt::QueuedConnection);
+			_previousValues.clear();
+		}
+		else
+		{
+			// If LEDDevice is not switchable, simulate OFF-state providing a continous stream of BLACK
+			this->switchOff();
+		}
+	}
 }
 
 void LinearColorSmoothing::setEnable(bool enable)
 {
+	LedDevice::setEnable(enable);
```

Comment:
Diese Anweisung sollte entfernt werden oder aus welchem Grund sollte das LedDevice ausgeschaltet werden, wenn LinearColorSmoothing ausgeschaltet wird?

Reply:
> Diese Anweisung sollte entfernt werden oder aus welchem Grund sollte das LedDevice ausgeschaltet werden, wenn LinearColorSmoothing ausgeschaltet wird?

Es geht hier nicht um das eigentliche LedDevice, sondern um das Smoothing-Device, welches als "Art" Device implementiert ist. Smooting erbt von LedDevice und damit das SmoothingDevice den richtigen Staus wird hier das enable Flag der Basisklasse gesetzt.

Wenn Du in LinearColorSmoothing.h schaust, siehst Du, dass genau dieses Flag heute schon ausgewertet wird. 
`bool enabled() { return LedDevice::enabled() && !_pause; };`
Nur wurde es vorher noch nicht definiert gesetzt....

Reply:
> Es geht hier nicht um das eigentliche LedDevice, sondern um das Smoothing-Device, welches als "Art" Device implementiert ist. Smooting erbt von LedDevice und damit das SmoothingDevice den richtigen Staus wird hier das enable Flag der Basisklasse gesetzt.

Genau. Nur das Problem wenn du dir die Reihenfolge vom Abschalten der Komponenten ansiehst, schaltet sich LinearColorSmoothing direkt nach Hyperion ab somit wird auch kein Schwarzwert mehr gesendet. Gerade ausprobiert.

Reply:
Es geht hier nicht um das Herunterfahren, sondern das man in UI Smoothing ausschaltet. Oder Hyperion disablest.
Am besten siehst Du die Effekte, wenn Du ein File-Device dranhängst.
Das kannst Du als Switchable Y/N über das UI konfigurieren. 
Dann noch Timestamps für das File (neu) anschalten und ein "tail" auf das File machen.

Da sieht man gut was im write wirklich ankommt....

Reply:
> wird hier das enable Flag der Basisklasse gesetzt

Und warum sollte LinearColorSmoothing beim Ausschalten die basisklasse ausschalten?
Verstehe ich nicht.

Reply:
> Es geht hier nicht um das Herunterfahren, sondern das man in UI Smoothing ausschaltet. Oder Hyperion disablest.

Ich habe nie was von herunterfahren geschrienen. Es geht hier klarer weise nur um das AUSSCHALTEN von komponenten.
Ich glaube wir reden an einander vorbei.

Reply:
Weil enabled() von LinearColorSmoothing diesen Wert abfragt (siehe oben).

d.h.

Setzen:
LedDevice::setEnable(enable);

Abfrage:

bool enabled() { return LedDevice::enabled() && !_pause; };

Reply:
> Weil enabled() von LinearColorSmoothing diesen Wert abfragt (siehe oben).

Das ist richtig. LinearColorSmoothing muss ja wissen ob deine LEDs auch eingeschaltet sind.

> Setzen:
> LedDevice::setEnable(enable);

Ja, dein LedDevice und nicht die LinearColorSmoothing.

So und nun zu dem Problem. Schaltest du nur die LinearColorSmoothing aus, wird das signal componentStateChanged von Hyperion ausgelöst und triggert deinen Slot componentStateChange von LinearColorSmoothing. Nachdem die "wenn" Bedingung
`if(component == hyperion::COMP_SMOOTHING)`
zutrifft, wird die funktion
`setEnable(state);`
im LinearColorSmoothing ausgeführt.
Und TADA was kommt als erstes in der funktion?
`LedDevice::setEnable(enable);`
Und wo führt das hin?
https://prnt.sc/pryx4t

Reply:
Hi
sorry ich weiß nicht was Du willst...
zuerst monierst Du die eingefügte  setEnable Anweisung in LinearColorSmoothing::setEnable

> Diese Anweisung sollte entfernt werden

Und dann erklärst Du mir das über diese Anweisung der enable-Zustand doch von LinearColorSmoothing gesetzt wird 

> Und TADA was kommt als erstes in der funktion?
> LedDevice::setEnable(enable);

Genau darum habe ich sie eingeführt. Was ist also jetzt das Problem?

Reply:
> Hi
> sorry ich weiß nicht was Du willst...
> zuerst monierst Du die eingefügte setEnable Anweisung in LinearColorSmoothing::setEnable
> 
> > Diese Anweisung sollte entfernt werden
> 
> Und dann erklärst Du mir das über diese Anweisung der enable-Zustand doch von LinearColorSmoothing gesetzt wird
> 
> > Und TADA was kommt als erstes in der funktion?
> > LedDevice::setEnable(enable);
> 
> Genau darum habe ich sie eingeführt. Was ist also jetzt das Problem?

Ich werde heute keine Diskussion mehr mit dir führen. Du verstehst mein anliegen sowieso nicht. Ich merge gerne deinen PR aber danach bin ich der erste der ein Issue eröffnet, weil meine Led Devices nicht mehr abschalten (bzw. auf schwarz schalten).

Reply:
Ah jetzt ja... 
heute Morgen um 5:00 Uhr hat es klick gemacht... wozu freie Kapazitäten und Abstand doch gut sind.

Warum sagst Du nicht, dass das im enable geworfene Signal nicht Smoothing, sondern das eigentliche Device zusätzlich beeinflusst... :) :) :)